src/helpers/helpers.core.ts

```diff
@@ -322,39 +375,25 @@ function _getKeyResolver(key) {
   };
 }
 
-/**
- * @private
- */
-export function _splitKey(key) {
-  const parts = key.split('.');
-  const keys = [];
-  let tmp = '';
-  for (const part of parts) {
-    tmp += part;
-    if (tmp.endsWith('\\')) {
-      tmp = tmp.slice(0, -1) + '.';
-    } else {
-      keys.push(tmp);
-      tmp = '';
-    }
-  }
-  return keys;
+export function resolveObjectKey(obj: AnyObject, key: string): AnyObject {
+  const resolver = keyResolvers[key] || (keyResolvers[key] = _getKeyResolver(key));
+  return resolver(obj);
 }
 
 /**
  * @private
  */
-export function _capitalize(str) {
+export function _capitalize(str: string) {
   return str.charAt(0).toUpperCase() + str.slice(1);
 }
 
 
-export const defined = (value) => typeof value !== 'undefined';
+export const defined = (value: unknown) => typeof value !== 'undefined';
 
-export const isFunction = (value) => typeof value === 'function';
+export const isFunction = (value: unknown) => typeof value === 'function';
 
 // Adapted from https://stackoverflow.com/questions/31128855/comparing-ecma6-sets-for-equality#31129384
-export const setsEqual = (a, b) => {
+export const setsEqual = (a: Set<unknown>, b: Set<unknown>) => {
```

Comment:
Would it better to type this as 

```ts
export const setsEqual = <T>(a: Set<T>, b: Set<T>) => 
```

That way the compiler would warn if we compare sets of different types

Reply:
@etimberg it was taken directly from [types/helpers/helpers.core.d.ts](https://github.com/chartjs/Chart.js/blob/master/types/helpers/helpers.core.d.ts#L157), but ok, i fixed it